### ISSUE TITLE
Is this note in max-len.md obsolete?

### ISSUE THREAD

**Issue by vsemozhetbyt** [eslint/eslint#6637]
max-len [doc](https://github.com/eslint/eslint/blob/master/docs/rules/max-len.md):

> Note: This rule calculates the length of a line via code points, not characters. That means if you use a double-byte character in your code, it will count as 2 code points instead of 1, and 2 will be used to calculate line length. This is a technical limitation of JavaScript that is made easier with ES2015, and we will look to update this when ES2015 is available in Node.js.

1. The 'code point vs character' here seems to be in conflict with ES6 terminology (compare `fromCodePoint` vs `fromCharCode`). May be something about UCS-2 or 16-bit shoud be added.

2. ES2015 is available in Node.js. So may be some notes about maintained compatibility with old versions should be added instead, if this is the cause.

**Comment by nzakas** [comment]
Hmm, it should be "code units" instead of "code points", but otherwise accurate. 

The ES2015 features in Node 4 aren't complete, so we can't remove this warning or make other changes.

**Comment by vsemozhetbyt** [comment]
Shoul I make a PR with "code points" replaced by "code units"? And my be "available in Node.js" could be replaced by "available in Node.js LTS"?

**Comment by kaicataldo** [comment]
@vsemozhetbyt I don't think we should change it to "available in Node.js LTS" because we will continue supporting Node 4 for the foreseeable future.

**Comment by vsemozhetbyt** [comment]
@kaicataldo Got it. And what about "code points" -> "code units"?

**Comment by kaicataldo** [comment]
Sounds like @nzakas is on board - I say go for it :) Here's some more info on making a PR: http://eslint.org/docs/developer-guide/contributing/pull-requests. Thanks for contributing!

**Comment by vsemozhetbyt** [comment]
@kaicataldo I am on it soon.

@nzakas @kaicataldo By the way, it seems the team can get string code point length in Node 4. The spread operator (`[...str].length`) or unicode RegExp (`str.split(/[^]{0}/u).length`) are not supported in Node.js 4, but this way I've just checked on Node.js 4.4.7 and it is OK:

```js
const str = '\ud83d\udc0e\ud83d\udc71\u2764';

const strLength = str.length;

let strUniLength = 0;
for (let codePoint of str) strUniLength++;

console.log(`strLength: ${strLength}, strUniLength: ${strUniLength}`);
```
```
strLength: 5, strUniLength: 3
```
